Update proxy instructions, add SAP example file (#232)
uyuni-project/uyuni-docs: modules/upgrade/pages/upgrade-script-example.adoc

= Sample Autoinstallation Script

This section contains example autoinstallation scripts.

If you use these profiles, please check which variables they use.
When the uploaded profile requires variables to be set, navigate to menu:Systems[Autoinstallation > Profiles], select the profile to edit, and navigate to the [guimenu]``Variables`` tab.
Specify the required variables, using this format:

----
<key>=<value>
----

For all installations, the ``$redhat_management_server`` variable will be set automatically and does not need to be defined.

> The examples uses often the ``$distrotree`` variable, which must be defined as the distribution label which is used with this profile.

Author: ```suggestion
These examples use the ``$distrotree`` variable, which must be defined as the distribution label which is used with this profile.
```

Set the variable to the same value as you select in ``Autoinstall Tree`` in the [guimenu]``Details`` tab.
Example:

----
distrotree=sles_sap15sp1-x86_64
----

> when the distribution label is `sles_sap15sp1-x86_64`.

Reviewer: ```suggestion
when the distribution label is [systemitem]``sles_sap15sp1-x86_64``.
```

== Upgrading from SLES{nbsp}12{nbsp}SPn to SLES{nbsp}15{nbsp}SP1

Use this script as the basis for an upgrade from SLES{nbsp}12{nbsp}SPn to SLES{nbsp}15{nbsp}SP1.

> Please check the official Product documentation to find the supported upgrade paths.

Reviewer: ```suggestion
Check the official {sles} documentation to find the supported upgrade paths.
```

Reviewer 2: I see this more general and not limited to SLES. So maybe
```suggestion
Check the official Product documentation to find the supported upgrade paths.
```

?

  </scripts>
</profile>
----

> 

Reviewer 2: The URLs in this profile are wrong. The last part of the path must be the same as it is the distribution label.

`summer2017` in a mix with `autumn2018` cannot work.

I would suggest we unify all example profiles and use a variable `$distrotree` like in the SAP example.

  </scripts>
</profile>
----

> == Upgrading a Proxy from SLE{nbsp}15 to SLE{nbsp}15.1

Reviewer 2: ```suggestion
== Upgrading a Proxy from 3.2 to 4.0
```
Or if you want to provide SLE versions, 12 SP4 to 15 SP1

> Use this script as the basis for an upgrade from SLE{nbsp}15 to SLE{nbsp}15.1.

Reviewer 2: Wrong, this example is for proxy

> .Example: Script for Upgrading from SLE{nbsp}12{nbsp}SP4 to SLE{nbsp}15

Reviewer 2: This is for proxy, not SLE

    <modified config:type="boolean">true</modified>
    <remove_old config:type="boolean">false</remove_old>
  </backup>
  <networking>
    <keep_install_network config:type="boolean">true</keep_install_network>
    <start_immediately config:type="boolean">true</start_immediately>
  </networking>
  <software>
    <image/>
    <install_recommended config:type="boolean">true</install_recommended>
    <instsource/>
    <patterns config:type="list">
      <pattern>suma_proxy</pattern>
    </patterns>
    <products config:type="list">
      <listentry>SUSE-Manager-Proxy</listentry>
    </products>
  </software>
  <scripts>
    <chroot-scripts config:type="list">
      <script>
        <filename>cleanup.sh</filename>
        <chrooted config:type="boolean">true</chrooted>
        <interpreter>shell</interpreter>
        <source>
        <![CDATA[
#!/bin/sh
         sysconf_addword /etc/sysconfig/apache2 APACHE_MODULES wsgi
         if [ -e /etc/rhn/rhn.conf.rpmsave ]; then
             cp /etc/rhn/rhn.conf.rpmsave /etc/rhn/rhn.conf
         fi
         zypper rr --all
         ]]>
        </source>
      </script>
    </chroot-scripts>
  </scripts>
</profile>
----

> == Upgrading an SAP Client from SLE{nbsp}15 to SLE{nbsp}15.1

Reviewer 2: The "from" is not so important. Maybe we just write "to" SLE 15 SP1?

> Use this script as the basis for an upgrade from SLE{nbsp}15 to SLE{nbsp}15.1.

Reviewer 2: Here you say "from 15" bug the example title is "from 12 SP4" .

.Example: Script for Upgrading an SAP Client from SLE{nbsp}12{nbsp}SP4 to SLE{nbsp}15

If you use this profile, when the uploaded profile requires variables to be set, navigate to menu:Systems[Autoinstallation > Profiles], select the profile to edit, and navigate to the [guimenu]``Variables`` tab.
Specify the required variables, using this format:

----
<key>=<value>
----

For SAP installations, the ``$redhat_management_server`` variable will be set automatically and does not need to be defined.

This example uses the ``$distrotree`` variable, which must be defined as the distribution label.
Set this variable as:

----
distrotree=sles_sap15sp1-x86_64

> ----

Reviewer 2: Should we put this in a general section? It is valid for all autoinstallations.

We should also say something about the registration at the end. Depending on what mechanism you use (salt vs. traditional) you need different snippets (minion_script vs sles_register_script).

        </source>
      </script>
    </chroot-scripts>
    <init-scripts config:type="list">
      $SNIPPET('spacewalk/minion_script')
<!-- when client is traditional registered, please use this snippet
      $SNIPPET('spacewalk/sles_register_script')
-->
    </init-scripts>
  </scripts>
</profile>
----

> == Upgrading an SAP Client from SLE{nbsp}12{nbsp}SPn to SLE{nbsp}15{nbsp}SP1

Author: I'm not sure it's clear what SPn would mean. Can we just say "SLE{nbsp}12" without specifying an SP?

Reviewer 2: I think you and @keichwa should come to a conclusion how many "version" we should name here.

The "source" does not matter. Only the target version is important as the examples migrate to a specific version.
So if we say just SLE12 or SLE12 SPn or nothing is something you need to decide.

Author: I would prefer to remove the versioning as much as possible from this entire guide, but that's for a different PR. For now, let's just leave it as SLE 12.

Use this script as the basis for an upgrade from SLE{nbsp}12{nbsp}SPn to SLE{nbsp}15.1.

> Please check the official Product documentation to find the supported upgrade paths.

Author: ```suggestion
Please check the official product documentation to find the supported upgrade paths.
```

Reviewer 2: The other place with this sentence was suggested to be:

```suggestion
Check the official product documentation to find the supported upgrade paths.
```
